src/AddressSpace.cc

```diff
@@ -1260,6 +1264,18 @@ void AddressSpace::unmap(Task* t, remote_ptr<void> addr, ssize_t num_bytes) {
     return;
   }
 
+  for (KernelMapIterator it(t); !it.at_end(); ++it) {
+    auto& km = it.current();
+    if (   !km.fsname().empty()
+        && !km.is_heap() && !km.is_stack() && !km.is_vvar()
+        && km.prot() & PROT_EXEC)
+    {
+      if (!executable_mappings_changed_ && km.prot() & PROT_EXEC) {
+        executable_mappings_changed_ = true;
+      }
```

Comment:
If we use ld.so's library list then I think we may need to set our own breakpoints in the ld.so debug hooks, which makes this a lot more work. But I think it's the right thing to do.

Reply:
I have to check, but I guess wine is not notifying ld.so when loading PE modules,
so there would be no benefit for target_wine?

Reply:
Ok.

Reply:
If the dll is a (with current wine less used) .so file then it
would be done by dynamic loader via dlopen_unix_dll / dlopen.
But I have not found something related to dynamic loader for native dlls.

This is an attempt to outline how winedbg gets notified about a dll loading:
```
ntdll:LdrLoadDll
ntdll:load_dll
ntdll:load_native_dll
ntdll:build_module - SERVER_START_REQ( load_dll )
server:DECL_HANDLER(load_dll)
server:generate_debug_event( current, DbgLoadDllStateChange, dll );
server:alloc_debug_event
server:fill_load_dll_event
winedbg:event_data_to_state_change
winedbg:NtWaitForDebugEvent - SERVER_START_REQ( wait_debug_event )
winedbg:DbgUiWaitStateChange
winedbg:WaitForDebugEvent
winedbg:handle_debug_event
winedbg:LOAD_DLL_DEBUG_EVENT
winedbg:dbg_load_module
dbghelp:SymLoadModuleExW
```
Unfortunately I do not see a hook that would be more suitable or to get the address to do something like `add_breakpoint(p, BKPT_INTERNAL)`.